test/widgets/set_status_test.dart

```diff
@@ -0,0 +1,552 @@
+          .page.isA<SetStatusPage>();
+      }
+
+      group('no status set, buttons are disabled', () {
+        testWidgets('emoji is added -> buttons are enabled', (tester) async {
+          prepareBoringImageHttpClient();
+
+          Route<dynamic>? currentRoute;
+          final testNavObserver = TestNavigatorObserver();
+          testNavObserver.onPushed = (route, _) => currentRoute = route;
+          testNavObserver.onPopped = (_, previous) => currentRoute = previous;
+
+          await setupPage(tester,
+            emojiData: serverEmojiDataPopular,
+            navigatorObserver: testNavObserver);
+
+          checkButtonEnabled(tester, clearButtonFinder, expected: false);
+          checkButtonEnabled(tester, saveButtonFinder, expected: false);
+
+          // Choose 'slight_smile' from popular emojis.
+          await chooseEmojiFromPicker(tester, '1f642',
+            navObserver: testNavObserver, currentRoute: () => currentRoute!);
+
+          checkButtonEnabled(tester, clearButtonFinder, expected: true);
+          checkButtonEnabled(tester, saveButtonFinder, expected: true);
+
+          debugNetworkImageHttpClientProvider = null;
+        });
+
+        testWidgets('text is added -> buttons are enabled', (tester) async {
+          await setupPage(tester);
+
+          checkButtonEnabled(tester, clearButtonFinder, expected: false);
+          checkButtonEnabled(tester, saveButtonFinder, expected: false);
+
+          await tester.enterText(findStatusTextField(), 'Happy');
+          await tester.pump();
+
+          checkButtonEnabled(tester, clearButtonFinder, expected: true);
+          checkButtonEnabled(tester, saveButtonFinder, expected: true);
+        });
+
+        testWidgets('empty spaces are added as text -> buttons stays disabled', (tester) async {
+          await setupPage(tester);
+
+          checkButtonEnabled(tester, clearButtonFinder, expected: false);
+          checkButtonEnabled(tester, saveButtonFinder, expected: false);
+
+          await tester.enterText(findStatusTextField(), '   ');
+          await tester.pump();
+
+          checkButtonEnabled(tester, clearButtonFinder, expected: false);
+          checkButtonEnabled(tester, saveButtonFinder, expected: false);
+        });
+
+        testWidgets('a suggestion is selected -> buttons are enabled', (tester) async {
+          await setupPage(tester, emojiData: suggestedEmojiData);
+
+          checkButtonEnabled(tester, clearButtonFinder, expected: false);
+          checkButtonEnabled(tester, saveButtonFinder, expected: false);
+
+          await tester.tap(findSuggestion(code: '1f6e0', text: 'Busy'));
+          await tester.pump();
+
+          checkButtonEnabled(tester, clearButtonFinder, expected: true);
+          checkButtonEnabled(tester, saveButtonFinder, expected: true);
+        });
+
+        testWidgets('emoji & text are added, then removed -> buttons are enabled, then disabled', (tester) async {
```

Comment:
There's a test just below for the buttons getting disabled on hitting clear; and other tests above in this group for them getting enabled in the first place on adding emoji or text. So it seems like everything this test is covering is already covered by more-specific tests, and it can be cut.

(Which is especially nice since it's the longest one in the group, at just over 30 lines.)